tensorflow/lite/c/c_api.cc

```diff
@@ -275,8 +275,16 @@ TfLiteInterpreter* InterpreterCreateWithOpResolver(
                                      error_reporter);
 
   std::unique_ptr<tflite::Interpreter> interpreter;
-  if (builder(&interpreter) != kTfLiteOk) {
-    return nullptr;
+  if (optional_options && optional_options->num_threads !=
+      TfLiteInterpreterOptions::kDefaultNumThreads) {
+    if (builder(&interpreter, optional_options->num_threads) !=
+        kTfLiteOk) {
+      return nullptr;
+    }
+  } else {
+    if (builder(&interpreter) != kTfLiteOk) {
+      return nullptr;
+    }
```

Comment:
Please cleanup the SetNumThreads in the following block.

Reply:
@dev0x13, I think this change isn't necessary. In C APIs, the number of threads (that's set via SetNumThreads in the following block) will be passed correctly to the default XNNPACK delegate because the AllocateTensors will be always called after the interpreter is created with C APIs (See my [latest comment](https://github.com/tensorflow/tensorflow/issues/42277#issuecomment-851178687) for a brief explanation).

Btw, in C APIs, as you can see by the definition of [tflite::CreateOpResolver()](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/create_op_resolver_with_builtin_ops.cc#L30), there'll be no default delegate (i.e. XNNPACK delegate) to be applied. To use XNNPACK delegate w/ C APIs, one has to explicitly add the XNNPACK delegate. Sorry for the confusion here.

Reply:
Sorry for the late reply. @multiverse-tf Thank you for the clarification! I am still working with 2.4.0 codebase, where `tflite::ops::builtin::BuiltinOpResolver` is used instead of `tflite::ops::builtin::BuiltinOpResolverWithoutDefaultDelegates`, so I missed this important change. So that this PR is actually unneeded now, so closing it.